NOTICE.md

```diff
@@ -28,12 +28,52 @@ The project maintains the following source code repositories:
 
 * https://github.com/eclipse-iceoryx/iceoryx
 
-## Third-party Content
+## Third-party Dependencies
 
 This project leverages the following third party content.
-
-None
-
+The corresponding license files can be found at https://github.com/eclipse-iceoryx/iceoryx/tree/master/doc/3rd_party_licenses or in the folder `share/doc/iceoryx/3rd_party_licenses` of your local iceoryx installation.
+
+### System Libraries
+libacl library
+ * Usage: as-is, dynamic-linking
+ * Notes: system-header `<sys/acl.h>` is used.
+
+libncurses library
+ * Usage: as-is, dynamic-linking
+ * Notes: system-header `<ncurses.h>` is used.
+
+### Build Dependencies
+cpptoml library (v0.1.1)
```

Comment:
They are required by default, but are not hard dependencies since they can be switched off. Can we mark that somehow? Like `* optional; default = on` or similar